lib/note-list/index.tsx

```diff
@@ -215,86 +215,53 @@ export class NoteList extends Component<Props> {
     this.toggleShortcuts(true);
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps: Props): void {
-    const {
-      notes,
-      noteDisplay,
-      openedTag,
-      selectedNote,
-      selectedNoteContent,
-      showNoteList,
-      showTrash,
-      tagResultsFound,
-    } = nextProps;
-    const { selectedIndex } = this.state;
-
+  componentDidUpdate(prevProps: Props) {
     if (
-      noteDisplay !== this.props.noteDisplay ||
-      notes !== this.props.notes ||
-      tagResultsFound !== this.props.tagResultsFound ||
-      selectedNoteContent !== this.props.selectedNoteContent ||
-      showNoteList !== this.props.showNoteList
+      prevProps.noteDisplay !== this.props.noteDisplay ||
+      prevProps.notes !== this.props.notes ||
+      prevProps.tagResultsFound !== this.props.tagResultsFound ||
+      prevProps.selectedNoteContent !== this.props.selectedNoteContent ||
+      prevProps.showNoteList !== this.props.showNoteList ||
+      prevProps.searchQuery !== this.props.searchQuery ||
+      prevProps.showTrash !== this.props.showTrash
     ) {
       heightCache.clearAll();
     }
 
-    // jump to top of note list
+    // reselect when a note is removed
     if (
-      openedTag !== this.props.openedTag ||
-      showTrash !== this.props.showTrash
+      !this.props.selectedNote &&
+      this.state.selectedIndex !== null &&
+      this.props.notes[this.state.selectedIndex]
     ) {
-      const hasNotes = notes.length > 0;
-      this.setState({ selectedIndex: hasNotes ? 0 : null });
-      this.props.onSelectNote(hasNotes ? notes[0] : null);
-      return;
+      this.props.onSelectNote(this.props.notes[this.state.selectedIndex]);
```

Comment:
does this still belong? I thought we got into a quick loop at times when the note would disappear from the sort selection and then reappear.